plugins/tile/tree.cpp

```diff
@@ -388,16 +410,17 @@ wf::geometry_t view_node_t::calculate_target_geometry()
     /* If view is maximized, we want to use the full available geometry */
     if (view->pending_fullscreen())
     {
-        auto vp = wset->get_current_workspace();
-        int view_vp_x = std::floor(1.0 * geometry.x / size.width);
-        int view_vp_y = std::floor(1.0 * geometry.y / size.height);
-
-        local_geometry = {
-            (view_vp_x - vp.x) * size.width,
-            (view_vp_y - vp.y) * size.height,
-            size.width,
-            size.height,
-        };
+        auto base_geometry = wset->get_attached_output()->get_relative_geometry();
+        local_geometry = adjust_geometry_for_workspace(wset, base_geometry, local_geometry);
+    } else if (this->show_maximized)
+    {
+        auto base_workarea = wset->get_attached_output()->workarea->get_workarea();
```

Comment:
same as above, in very bad cases, get_attached_output() is null, we need a check for this (in that case we can probably just skip showing maximized, if the wset is not attached to an output, the user doesn't see it anyway)